### ISSUE TITLE
Wrong encoding of file

### ISSUE THREAD

**Issue by SebastianOpriel** [FraunhoferISST/DataspaceConnector#755]
**Describe the bug**
I try to create a resource with an publicly available weather API. When I access the data URL, a malformed payload is downloaded. I will provide you the file on request , because I can't determine what the payload is due to a wrong encoding.

**To Reproduce**
1. Checked out last version of Master (22.11.2021 10:20)
2. Configured for VM deployment of a hoster (PostgreSQL DB, Productive_mode, DAPS, passwords, certificates etc.)
3. Run it via docker-compose up
4. Configure the offering via SwaggerUI:

Catalog
```
{
  "title": "Demo",
  "description": "Demo"
}
```

Resource:
```
{
  "title": "Wetterdaten aus Dortmund",
  "description": "Wetterdaten aus Dortmund",
  "keywords": [
    "weather"  ],
  "publisher": "https://domain.de",
  "language": "EN",
  "license": "https://domain.de",
  "sovereign": "https://domain.de",
  "endpointDocumentation": "https://domain.de",
  "paymentMethod": "undefined"
}
```

Representation:
```
{
  "title": "Weather Representation",
  "description": "Weather Representation",
  "mediaType": "json",
  "language": "EN"
}
```

Artifact:
```
{
  "title": "Weather Data from Dortmund",
  "description": "Weather Data from Dortmund",
  "accessUrl": "https://samples.openweathermap.org/data/2.5/weather?q=Dortmund&appid=439d4b804bc8187953eb36d2a8c26a02",
  "automatedDownload": false
}
```

Contract:
```
{
  "title": "General Contract",
  "description": "General Contract",
  "consumer": null,
  "provider": null,
  "start": "2021-11-23T09:00:53.195Z",
  "end": "2021-12-23T09:00:53.195Z"
}
```

Rule:
```
{
  "title": "5-times rule",
  "description": "5-times",
  "value": "{   \"@context\" : {     \"xsd\" : \"http://www.w3.org/2001/XMLSchema#\",     \"ids\" : \"https://w3id.org/idsa/core/\",     \"idsc\" : \"https://w3id.org/idsa/code/\"   },   \"@type\" : \"ids:Permission\",   \"@id\" : \"https://w3id.org/idsa/autogen/permission/27ba70fd-85d8-4420-b1f7-0460eed273b3\",   \"ids:description\" : [ {     \"@value\" : \"n-times-usage\",     \"@type\" : \"http://www.w3.org/2001/XMLSchema#string\"   } ],   \"ids:title\" : [ {     \"@value\" : \"Example Usage Policy\",     \"@type\" : \"http://www.w3.org/2001/XMLSchema#string\"   } ],   \"ids:assignee\" : [ ],   \"ids:assigner\" : [ ],   \"ids:postDuty\" : [ ],   \"ids:constraint\" : [ {     \"@type\" : \"ids:Constraint\",     \"@id\" : \"https://w3id.org/idsa/autogen/constraint/d111a0ee-8ef8-42b1-a773-9e72e46dcdc0\",     \"ids:leftOperand\" : {       \"@id\" : \"https://w3id.org/idsa/code/COUNT\"     },     \"ids:rightOperand\" : {       \"@value\" : \"5\",       \"@type\" : \"xsd:double\"     },     \"ids:operator\" : {       \"@id\" : \"https://w3id.org/idsa/code/LTEQ\"     }   } ],   \"ids:action\" : [ {     \"@id\" : \"https://w3id.org/idsa/code/USE\"   } ],   \"ids:preDuty\" : [ ] }"
}
```

5. Try to access the data via API: https://domain.de/api/artifacts/2253f075-a17a-443a-a4b8-a7b3f18bbb2b/data
6. A file is downloaded with a wrong encoding.

Further Context:
In the logs I see an Exception before the data are pulled from https://samples.openweathermap.org:
```
2021-11-23T09:43:17,060 [https-jsse-nio-8080-exec-7] DEBUG - 
org.springframework.jdbc.datasource.DataSourceUtils:201 - Could not set JDBC Connection read-only
org.postgresql.util.PSQLException: Cannot change transaction read-only property in the middle of a transaction.
at org.postgresql.jdbc.PgConnection.setReadOnly(PgConnection.java:785) ~[postgresql-42.3.1.jar!/:42.3.1]
[...]
at io.dataspaceconnector.service.resource.base.BaseEntityService.get(BaseEntityService.java:105) [classes!/:6.5.1]
at io.dataspaceconnector.service.resource.type.ArtifactService.getData(ArtifactService.java:266) [classes!/:6.5.1]
at io.dataspaceconnector.controller.resource.type.ArtifactController.getData(ArtifactController.java:156) [classes!/:6.5.1]
[...]
```

After that the URL is requested:
```
2021-11-23T09:43:17,068 [https-jsse-nio-8080-exec-7] INFO - de.fraunhofer.ids.messaging.protocol.http.IdsHttpService:393 - Sending request to https://samples.openweathermap.org/data/2.5/weather?q=Dortmund&appid=439d4b804bc8187953eb36d2a8c26a02 ...
2021-11-23T09:43:17,175 [https-jsse-nio-8080-exec-7] INFO - de.fraunhofer.ids.messaging.protocol.http.IdsHttpService:405 - Successfully received response to request.
```

**Comment by SebastianOpriel** [comment]
What I observed now: The samples API doesn't set the content type.
`https://samples.openweathermap.org/data/2.5/weather?q=Dortmund&appid=439d4b804bc8187953eb36d2a8c26a02`

![image](https://user-images.githubusercontent.com/22075788/143916221-d0d14357-8f59-4e84-b7e3-18860f3625b1.png)

In contrast I used the URI of a publicly available demo from the main page of https://openweathermap.org/ which's response contains the correct content type:
`https://openweathermap.org/data/2.5/weather?id=2935517&appid=439d4b804bc8187953eb36d2a8c26a02`

![image](https://user-images.githubusercontent.com/22075788/143916588-449876ca-7304-4609-9a35-869ecd13a334.png)

But this doesn't affect the output of a malformed body. If I use this API to receive a JSON response, everything works fine.
`https://api.plos.org/search?q=title:%22Drosophila%22%20and%20body:%22RNA%22&fl=id,abstract&wt=json&indent=on`
So the bug might be a thing related to the openweatherapi

**Comment by juliapampus** [comment]
I could locate the error. As soon as an http response contains the header `Content-Encoding: gzip`, the body is not encrypted correctly. In the `HttpService`, we make use of the `de.fraunhofer.ids.messaging.protocol.http.HttpService`. It seems that there, the default unzip interceptor is missing.

I tried to add the header `accept-encoding: gzip` to the request, what should work:
![image](https://user-images.githubusercontent.com/72392527/146389929-3cfea44d-4e5d-4ed9-9880-dbbbf60dea35.png)

However, is does not:
![image](https://user-images.githubusercontent.com/72392527/146390222-db922345-b802-4661-ade8-e065a47e524f.png)

Would be great if you could take a look at this issue @tmberthold.

**Comment by juliapampus** [comment]
I also tried to add an unzip method (see [here](https://stackoverflow.com/questions/51901333/okhttp-3-how-to-decompress-gzip-deflate-response-manually-using-java-android)) to read the response correctly. This works for e.g. https://samples.openweathermap.org/data/2.5/weather?q=Dortmund&appid=439d4b804bc8187953eb36d2a8c26a02. So this gzip thing is definitely the error. Nevertheless, this does not work for e.g. https://openweathermap.org/. I think this has to happen in the Messaging Services by adding this to the http client, not by modifying the response. @tmberthold

**Comment by ghost** [comment]
Apparently we haven't had the use case yet, will look into it.

**Comment by brianjahnke** [comment]
Hi,
seems to be related. [IDS DeploymentExamples](https://github.com/International-Data-Spaces-Association/IDS-Deployment-Examples/pull/16#issuecomment-996013132)

**Comment by ghost** [comment]
Added a PR introducing a manual EncodingInterceptor to the OKHttpClient used. Still needs to be validated, if it fixes the problem.

https://github.com/International-Data-Spaces-Association/IDS-Messaging-Services/pull/403

**Comment by ghost** [comment]
I can confirm that with the manually added Interceptor, the three APIs mentioned above all work now without problems at least in my tests using the current Dataspace Connector main (Changelog: 7.0.0) and my test-connector.

Tested APIs:

- [X] https://samples.openweathermap.org/data/2.5/weather?q=Dortmund&appid=439d4b804bc8187953eb36d2a8c26a02
- [X] https://openweathermap.org/data/2.5/weather?id=2935517&appid=439d4b804bc8187953eb36d2a8c26a02
- [X] https://api.plos.org/search?q=title:%22Drosophila%22%20and%20body:%22RNA%22&fl=id,abstract&wt=json&indent=on

I have also not noticed any problems interacting with other IDS-components due to the new Interceptor (e.g. DAPS & ClearingHouse).

Interceptor will be available with the next Messaging-Services version. If further adjustments are necessary, just let me know.

**Comment by juliapampus** [comment]
Solved with #846
